tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,413 @@
+__author__ = "Joon Bang"
+__status__ = "Prototype"
+
+INPUT_FILE = "tex2wiki/data/01outb.tex"
+OUTPUT_FILE = "tex2wiki/data/01outb.mmd"
+GLOSSARY_LOCATION = "tex2wiki/data/new.Glossary.csv"
+METADATA_TYPES = ["substitution", "constraint"]
+METADATA_MEANING = {"substitution": "Substitution(s)", "constraint": "Constraint(s)"}
+
+import csv
+
+
+class LatexEquation(object):
+    def __init__(self, label, equation, metadata):
+        self.label = label
+        self.equation = equation
+        self.metadata = metadata
+
+
+def generate_html(tag_name, text, options=None, spacing=2):
+    # (str, str(, dict, bool)) -> str
+    """
+    Generates an html tag, with optional html parameters.
+    When spacing = 0, there should be no spacing.
+    When spacing = 1, the tag, text, and end tag are padded with spaces.
+    When spacing = 2, the tag, text, and end tag are padded with newlines.
+    """
+    if options is None:
+        options = {}
+
+    option_text = [key + "=\"" + value + "\"" for key, value in options.iteritems()]
+    result = "<" + tag_name + " " * (option_text != []) + ", ".join(option_text) + ">"
+
+    if spacing == 2:
+        result += "\n" + text + "\n</" + tag_name + ">\n"
+    elif spacing == 1:
+        result += " " + text + " </" + tag_name + ">"
+    else:
+        result += text + "</" + tag_name + ">"
+
+    return result
+
+
+def generate_math_html(text, options=None, spacing=True):
```

Comment:
I think I've tried that, but it was pretty unreadable. generate_math_html is just different enough to make it a separate function, rather than an option of generate_html.